src/test/java/org/openrewrite/staticanalysis/EqualsAvoidsNullTest.java

```diff
@@ -37,26 +37,68 @@ void invertConditional() {
           //language=java
           java(
             """
+              import java.util.stream.Stream;
               public class A {
+                  public static final String EXTERNAL_KEY = "EXTERNAL_KEY";
                   {
                       String s = null;
-                      if(s.equals("test")) {}
-                      if(s.equalsIgnoreCase("test")) {}
-                      System.out.println(s.compareTo("test"));
-                      System.out.println(s.compareToIgnoreCase("test"));
-                      System.out.println(s.contentEquals("test"));
+                      if (s.equals(EXTERNAL_KEY)) {}
+                      if (s.equalsIgnoreCase(EXTERNAL_KEY)) {}
+                      System.out.println(s.compareTo(EXTERNAL_KEY));
+                      System.out.println(s.compareToIgnoreCase(EXTERNAL_KEY));
+                      System.out.println(s.contentEquals(EXTERNAL_KEY));
+                      System.out.println(Stream.of(EXTERNAL_KEY)
+                                               .filter(item -> item.contentEquals(EXTERNAL_KEY))
+                                               .findFirst());
+                  }
+
+                  boolean isFoo(final String test) {
+                      return new B().getBar(EXTERNAL_KEY).contentEquals(EXTERNAL_KEY)
+                             || B.getBar2(EXTERNAL_KEY).contentEquals(EXTERNAL_KEY)
+                             || test.contentEquals(EXTERNAL_KEY);
+                  }
+              }
+
+              public static class B {
+                  String getBar(final String test) {
+                      return null;
+                  }
+
+                  static String getBar2(final String test) {
+                      return null;
```

Comment:
Instead of expanding the small document example it's probably best to start a new unit test for the exact problem you want to replicate. We try to keep the document example small for that precise purpose: show a new issue in isolation, and only that particular case, as that then also helps stepping through with the debugger.